### PR TITLE
Scarb expand initial impl

### PR DESCRIPTION
**Pull request by maciektr**
commit-id:8273b556

---

**Stack**:
- #1348
- #1347
- #1344 ⬅

⚠️ *Part of a stack created by [spr](https://github.com/ejoffe/spr). Do not merge manually using the UI - doing so may have unexpected results.*

### CODE REVIEW

File: scarb/src/ops/expand.rs [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+        .diagnostics_reporter
+        .ensure(&db)
+        .map_err(|err| err.into())
+        .map_err(|err| {
+            if !suppress_error(&err) {
+                ws.config().ui().anyhow(&err);
+            }
+
+            anyhow!("could not check `{package_name}` due to previous error")
+        })?;
```
Comment (mkaput) [inline scarb/src/ops/expand.rs:62]: LS will need to expand macros on potentially erroring projects

File: scarb/src/ops/expand.rs [hunk 2]
```diff
@@ -0,0 +1,214 @@
...
+    do_expand(&db, &compilation_unit, opts, ws)?;
+
+    Ok(())
+}
+
+/// Memorize opened modules for adding appropriate bracketing to the code.
+struct ModuleStack(Vec<String>);
+
+impl ModuleStack {
+    pub fn new() -> Self {
```
Comment (mkaput) [inline scarb/src/ops/expand.rs:73]: If type is private, then you do not need to make its methods `pub`

File: scarb/src/ops/expand.rs
Comment (mkaput) [inline scarb/src/ops/expand.rs:1]: any tests?

**Review by THenry14 — APPROVED** [review]
lgtm

File: scarb/src/ops/expand.rs (hunk 1 above)
Comment (maciektr) [reply scarb/src/ops/expand.rs:62]: I will change it in subsequent PR.

**Pull request merged**